src/main/java/com/springweb/dv_spring_web_mongo/logging/ProjectLoggingAspect.java

```diff
@@ -0,0 +1,56 @@
+package com.springweb.dv_spring_web_mongo.logging;
+
+import com.springweb.dv_spring_web_mongo.model.User;
+import com.springweb.dv_spring_web_mongo.service.ProjectService;
+import lombok.RequiredArgsConstructor;
+import org.aspectj.lang.ProceedingJoinPoint;
+import org.aspectj.lang.annotation.Around;
+import org.aspectj.lang.annotation.Aspect;
+import org.aspectj.lang.annotation.Pointcut;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.stereotype.Component;
+import org.springframework.util.StopWatch;
+
+import javax.servlet.http.HttpServletRequest;
+
+@Component
+@Aspect
+@RequiredArgsConstructor
+public class ProjectLoggingAspect {
+
+    private final HttpServletRequest request;
+    private Logger logger = LoggerFactory.getLogger(ProjectService.class);
+
+    @Pointcut("@annotation(org.springframework.web.bind.annotation.GetMapping)" +
+            "|| @annotation(org.springframework.web.bind.annotation.PostMapping)" +
+            "|| @annotation(org.springframework.web.bind.annotation.PutMapping)" +
+            "|| @annotation(org.springframework.web.bind.annotation.DeleteMapping)")
+    public void anyRequestInProjectService() {
+    }
+
+    @Around("anyRequestInProjectService()")
+    public void afterServiceMethod(ProceedingJoinPoint joinPoint) throws Throwable {
+        logger.info("----------INCOMING REQUEST----------");
+        logger.info("HTTP request type: " + request.getMethod());
+        logger.info("Request URI: " + request.getRequestURI());
+        logger.info("Client IP: " + request.getRemoteAddr());
+        Authentication authentication = SecurityContextHolder.getContext().getAuthentication();
+        if (authentication.isAuthenticated()) {
+            User user = (User) authentication.getPrincipal();
```

Comment:
1. Please check how it works for public method (POST /api/users/register) without authorization. But now the method is not public because ther's one small mistake in the security config. Please fix it too.
2. Now it is not needed to cast `authentication.getPrincipal()` to `User` object. You can just call `authentication.getName()`